port/unix/omrsysinfo.c

```diff
@@ -2004,6 +2005,72 @@ omrsysinfo_get_OS_version(struct OMRPortLibrary *portLibrary)
 					rc = sysctlbyname(KERN_OSPRODUCTVERSION, buffer, &resultSize, NULL, 0);
 				}
 				buffer[resultSize] = '\0';
+
+				/* 10.16 is the version returned when in a backwards compatibility context in OSX versions 11+
+				* so we check the plink symlink as that bypasses the compatibility context and returns the
+				* real version info.
+				*/
+				if (0 == strncmp(buffer, "10.16", 5)) {
+					portLibrary->mem_free_memory(portLibrary, buffer);
```

Comment:
This should first check buffer != myBuffer, or free allocatedFileBuffer instead using the same comment as below where allocatedFileBuffer is freed. Also set allocatedFileBuffer to NULL so it can't be freed twice. allocatedFileBuffer  should be set to NULL above as well where it's freed.

Reply:
Freed allocatedFileBuffer here.

Reply:
`buffer` is NUL-terminated; shouldn't this use `strcmp()` instead?